src/mono/mono/mini/debugger-agent-external.c

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+#include <glib.h>
+#include <mono/metadata/components.h>
+#include "debugger-agent-external.h"
+
+#ifndef DISABLE_SDB
+
+void
+mono_debugger_agent_register_transport (DebuggerTransport *trans)
+{
+	mono_component_debugger ()->register_transport (trans);
```

Comment:
I'm a little bit worried this is going to be called too early - before we try to load the components.  We should verify that xamarin-macios can call this.

If it doesn't work, we should leave a list of DebuggerTransport  in the runtime, and have a function that the debugger can call to get the registered transports.